streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -190,6 +193,7 @@ public int process() {
                 requiresPoll = true;
             }
         } finally {
+            processorContext.setRecordContext(null);
```

Comment:
As it is the same behaviour it was prior to this change. Previously the `currRecord` would be set to `null` and `ProcessorContextImpl` would do `task.record()` to get this information and throw i.e.,
```
  if (task.record() == null)
            throw new IllegalStateException("This should not happen as topic() should only be called while a record is processed");
```
So it indicates that no records is currently being processed